devicetypes/qubino/qubino-flush-shutter.src/qubino-flush-shutter.groovy

```diff
@@ -0,0 +1,501 @@
+	switch (preference.type) {
+		case "enum":
+			return String.valueOf(integerValue)
+		default:
+			return integerValue
+	}
+}
+
+private getCommandValue(preference) {
+	def parameterKey = preference.key
+	switch (preference.type) {
+		case "range":
+			return settings."$parameterKey"
+		default:
+			return Integer.parseInt(settings."$parameterKey")
+	}
+}
+
+private isPreferenceChanged(preference) {
+	if (settings."$preference.key" != null) {
+		def value = state.currentPreferencesState."$preference.key"
+		return state.currentPreferencesState."$preference.key".value != settings."$preference.key"
+	} else {
+		return false
+	}
+}
+
+def handleConfigurationChange(confgurationReport) {
+	switch (confgurationReport.parameterNumber) {
+		case 71: //Operating mode
+			switch (confgurationReport.scaledConfigurationValue) {
+				case 0: // Shutter
+					checkAndTriggerModeChange("windowShade")
+					break
+				case 1: // Venetian
+					checkAndTriggerModeChange("windowShadeVenetian")
+					break
+			}
+			log.info "Current device's mode is: ${state.currentMode}"
+			break
+		case 72:
+			state.timeOfVenetianMovement = confgurationReport.scaledConfigurationValue
+			break
+		default:
+			log.info "Parameter no. ${confgurationReport.parameterNumber} has no specific handler"
+			break
+	}
+}
+
+private checkAndTriggerModeChange(reportedMode) {
+	if (state.currentMode != reportedMode) {
+		state.currentMode = reportedMode
+		createVenetianBlindsChildDeviceIfNeeded()
+	}
+}
+
+def parse(String description) {
+	def result = null
+	def cmd = zwave.parse(description)
+	if (cmd) {
+		result = zwaveEvent(cmd)
+	} else {
+		log.warn "${device.displayName} - no-parsed event: ${description}"
+	}
+	log.debug "Parse returned: ${result}"
+	return result
+}
+
+def multilevelChildInstalled(childDni) {
+	state.timeOfVenetianMovement = 150
+	sendHubCommand(encap(zwave.switchMultilevelV3.switchMultilevelGet(), 2))
+}
+
+def close() {
+	setShadeLevel(0x64)
+}
+
+def open() {
+	setShadeLevel(0x00)
+}
+
+def pause() {
+	sendHubCommand(encap(zwave.switchMultilevelV3.switchMultilevelStopLevelChange()))
+}
+
+def setLevelChild(level, childDni) {
+	setSlats(level)
+}
+
+def setLevel(level) {
+	state.blindsLastCommand = currentLevel > level ? "opening" : "closing"
+	setShadeLevel(level)
+}
+
+def setShadeLevel(level) {
+	log.debug "Setting shade level: ${level}"
+	def currentLevel = Integer.parseInt(device.currentState("level").value)
+	state.blindsLastCommand = currentLevel > level ? "opening" : "closing"
+	state.shadeTarget = level
+	encap(zwave.switchMultilevelV3.switchMultilevelSet(value: Math.min(0x63, level)))
+}
+
+def setSlats(level) {
+	def time = (int) (state.timeOfVenetianMovement  * 1.1)
+	sendHubCommand([
+			encap(zwave.switchMultilevelV3.switchMultilevelSet(value: Math.min(0x63, level)), 2),
+			"delay ${time}",
+			encap(zwave.switchMultilevelV3.switchMultilevelGet(), 2)
+	])
+}
+
+def refresh() {
```

Comment:
I think the usual pattern is that `refresh` just returns the command, but `ping` should actually wrap with `response()`, right?